### PR TITLE
Requesting pull for new feature/assignment7.Please review the same 

### PR DESCRIPTION
**Pull request by NitinSaxenait**
What does this change do?
Changes are added for new feature/assignment7 

Any additional information for the reviewer to start
NA

How should this be manually tested?
We need window OS in which Rust is installed. then execute this rust program

- [ x] Are there any changes pending?

No

Does any team have to be notified of changes in this feature?
No, only for the mentor.

Definition of Done:

- [x] - Is there >80% unit test code coverage?No

if not please mention the Code Coverage of code.

|| src/creating_substring.rs: 12/12 +0%
|| src/lib.rs: 0/1 +0%
|| src/pattern_matching.rs: 14/14 +0%
|| src/ques3.rs: 26/26 +0%
|| src/test.rs: 27/27 +0%
|| 
81.44% coverage, 79/97 lines covered, +0% change in coverage

- [ x] Will this feature require a new piece of infrastructure to be implemented?

- [x ] Is there appropriate logging included?yes

- [x] Does the project compile ok?Yes

- [ x] -Have Clippy violations been fixed?Yes

- [ x] Have Code is properly formatted?Yes

### CODE REVIEW

File: Assignments/assignment7/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+mod creating_substring;
+mod pattern_matching;
+mod ques3;
```
Comment (Dheerajjnv) [inline Assignments/assignment7/src/lib.rs:3]: Use a meaningful name.

**Review by Dheerajjnv — APPROVED** [review]

File: Assignments/assignment7/src/creating_substring.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+//
+// this function is generating subsets of a string.
```
Comment (ankit-mogha) [inline Assignments/assignment7/src/creating_substring.rs:2]: use a good comment "This" -> function name

File: Assignments/assignment7/src/lib.rs (hunk 1 above)
Comment (NitinSaxenait) [reply Assignments/assignment7/src/lib.rs:3]: okay

**Review by ankit-mogha — APPROVED** [review]

**Comment by PankajChaudhary5** [comment]
Test coverage is less than 90%
@Dheerajjnv @ankit-mogha

**Pull request closed without merging**